schemas/create-task-request.yml

```diff
@@ -44,6 +44,34 @@ properties:
       property isn't specified.
     type:           string
     pattern:        {$const: slugid-pattern}
+  dependencies:
+    title:          "Task Dependencies"
+    description: |
+      List of dependent tasks. These must either be _completed_ or _resolved_
+      before this task is scheduled. See `dependencyRelation` for semantics.
+    type:           array
+    default:        []
+    items:
+      title:        "Task Dependency"
+      description: |
+        The `taskId` of a task that must be resolved before this task is
+        scheduled.
+      type:         string
+      pattern:      {$const: slugid-pattern}
+    maxItems:       {$const: max-task-dependencies}
+    uniqueItems:    true
+  dependencyRelation:
+    title:          "Dependency Relation"
+    description: |
+      The tasks relation to its dependencies. This property specifies the
+      semantics of the `task.dependencies` property.
+      If `on-completed` is given the task will be scheduled when all
+      dependencies are resolved _completed_ (successful resolution).
+      If `on-resolved` is given the task will be scheduled when all dependencies
+      have been resolved, regardless of what their resolution is.
+    type:           string
+    enum:           {$const: dependency-relation}
+    default:        'on-completed'
```

Comment:
Do we really need the `on-` part of these? Maybe just `completed` and `resolved`. I don't really care either way, it just seems more straightforward that way.